tools/nodejs_api/src_cpp/node_progress_bar_display.cpp

```diff
@@ -14,8 +12,8 @@ void NodeProgressBarDisplay::updateProgress(uint64_t queryID, double newPipeline
     uint32_t oldPipelineProgress = (uint32_t)(pipelineProgress * 100.0);
     if (curPipelineProgress > oldPipelineProgress ||
```

Comment:
I looked into it a bit. The callback functions are wrapped with `Napi::ThreadSafeFunction`, which should automatically make the callback invocation on the main thread of Node.js runtime. However, it seems that the thread which calls the `ThreadSafeFunction` requires explict `Acquire ` and `Release` as per https://github.com/nodejs/node-addon-api/blob/main/doc/threadsafe_function.md#acquire which we did not seem to have.

Reply:
Looks like `Acquire` and `Release` are for reference counting (the `ThreadSafeFunction` will be destroyed when the count reaches 0). I don't think we ever call `Release()` on the `ThreadSafeFunction` so I don't think we need to worry about `ThreadSafeFunction` being destroyed while we still need it so I don't think we need to ever increment the reference count with `Acquire`.

I'm not sure if this is leaking memory though since I don't know if `ThreadSafeFunction` frees its resources in its destructor. I did notice a crash when I tried adding an explicit `Release` call to the destructor of `ConnectionQueryAsyncWorker` though so maybe it does. Regardless this is unrelated to the changes in this PR so I'll go ahead and merge this.

Reply:
OK.